File: ios/AMP/AMPCommon/AMPSRConfig.swift
//
//  AMPSRConfig.swift
//  AMPSample
//
//  Created by Boris BORGOBELLO on 10/12/2019.
//  Copyright © 2019 Streamroot. All rights reserved.
//

import Foundation
import StreamrootSDK

struct AMPSRConfig {
  static let ampLicense = {
    guard let path = Bundle.main.path(forResource: "amp", ofType: "lic") else { return nil }
    guard let licenseStr = try? String(contentsOfFile: path, encoding: .utf8), licenseStr.count > 0 else { return nil }
    return licenseStr.replacingOccurrences(of: "\n", with: "").replacingOccurrences(of: "\r", with: "")
  }() ?? "or put your AMP base64 licence here"
  
  static let url = "http://wowza-test.streamroot.io/liveOrigin/Sintel1/playlist.m3u8"
  static let dnaEnabled = true
  static let showStatsView = true
  
  static func setupDNA(_ builder: DNATrigger) -> DNATrigger {
    return builder
    .streamrootKey("demoswebsiteandpartners")
    .latency(30)
    .logLevel(.debug)
  }
}

File: android/AMP/app/src/main/java/io/streamroot/dna/samples/amp/AMPSRConfig.kt
package io.streamroot.dna.samples.amp

import android.content.Context
import io.streamroot.dna.core.Configure
import io.streamroot.dna.core.log.LogLevel
import java.io.BufferedReader
import java.io.InputStreamReader

object AMPSRConfig {
    fun AMP_LICENSE(context: Context) = {
        var inputStream: BufferedReader? = null
        var license: String?
        try {
            inputStream = BufferedReader(InputStreamReader(context.assets.open("amp.lic")))
            license = inputStream.readLine()
        } finally {
            inputStream?.close()
        }
        if (license == null || license.isEmpty()) null else license
    }() ?: "or put your license here"

    const val DEFAULT_VIDEO_URL = "http://wowza-test.streamroot.io/liveOrigin/Sintel1/playlist.m3u8"
    const val DNA_ENABLED = true

    // You can build DNA instance here
    fun configureStreamroot(baseConfig: Configure) = baseConfig.apply {
        latency(30)
        logLevel(LogLevel.DEBUG)
    }
}